<!--
  1 failure:  :disappointed_rel...
  0 warning: 
  1 messages
  
  DangerID: danger-id-default;
-->

<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  :disappointed_relieved: `npm test` failed. See below for details.
  </td>
    </tr>
  </tbody>
</table>



<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  ```
> titanium-mobile@8.0.0 test /Users/build/jenkins/workspace/ium-sdk_titanium_mobile_PR-10517
> grunt

Running "appcJs:src:lintOnly" (appcJs) task

Running "eslint:src" (eslint) task

/Users/build/jenkins/workspace/ium-sdk_titanium_mobile_PR-10517/android/modules/ui/src/js/listview.js
  120:20  warning  eval with argument of type CallExpression  security/detect-eval-with-expression
  120:20  error    eval can be harmful                        no-eval

✖ 2 problems (1 error, 1 warning)

Warning: Task "eslint:src" failed. Use --force to continue.

Aborted due to warnings.
npm ERR! Test failed.  See above for more details.

```
  </td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

--- a/android/modules/ui/src/js/listview.js
+++ b/android/modules/ui/src/js/listview.js
@@ -115,7 +115,18 @@ exports.bootstrap = function (Titanium) {
 					// widget does not exist, attempt to load namespace
 					widget = Module.main.require(namespace);
 				} catch (err) {
-					// do nothing...
+					// namespace does not exist, fall back to legacy behaviour
+					const namespaceIndex = namespace.lastIndexOf('.');
+					const proxyName = namespace.slice(namespaceIndex + 1);
+					const parentNamespace = namespace.substring(0, namespaceIndex);
+					const segments = parentNamespace.split('.');
+					let parentProxy = global;
+					for (let i = 0; i < segments.length; i++) {
+						parentProxy = parentProxy[segments[i]];
+					}
+					if (parentProxy) {
+						return parentProxy['create' + proxyName];
+					}
 				}
 			}
 			if (widget) {

